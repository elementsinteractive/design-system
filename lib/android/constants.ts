import { join } from 'path'

export const TEMPLATES = {
  typography: {
    source: join(__dirname, './templates/android/typographies.ejs'),
    destination: join(
      __dirname,
      '../shared/presentation/src/main/java/nl/elements/skeleton/presentation/compose/theme/Type.kt',
    ),
    filter: 'isTypography',
    formatter: 'android/compose/typography',
  },
  spacings: {
    source: join(__dirname, './templates/android/spacings.ejs'),
    destination: join(
      __dirname,
      '../shared/presentation/src/main/java/nl/elements/skeleton/presentation/compose/theme/Spacings.kt',
    ),
    filter: 'isSpacing',
    formatter: 'android/spacings',
  },
  colors: (destination: string) => ({
    source: join(__dirname, './templates/android/colors.ejs'),
    destination: `${destination}/Colors.kt`,
    filter: 'color',
    formatter: 'android/compose/colors',
  }),
  header: join(__dirname, './templates/android/header.ejs'),
}
