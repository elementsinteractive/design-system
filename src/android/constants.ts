import { join } from 'path'

export const TEMPLATES = {
  FILES: {
    header: join(__dirname, './templates/header.ejs'),
    colorTokens: join(__dirname, './templates/colorTokens.ejs'),
    colorTokensTypes: join(__dirname, './templates/colorTokensTypes.ejs'),
    spacingsTokens: join(__dirname, './templates/spacingsTokens.ejs'),
    spacingsTokensTypes: join(__dirname, './templates/spacingTokensTypes.ejs'),
    typographies: join(__dirname, './templates/typographies.ejs'),
    typographiesTypes: join(__dirname, './templates/typographiesTypes.ejs'),
  },
  FILTERS: {
    typography: 'isTypography',
    colors: 'isColor',
    sizings: 'isSizing',
    spacings: 'spacings',
  },
  FORMATTERS: {
    colorTokens: 'android/compose/colorTokens',
    colorTokensTypes: 'android/compose/colorTokenTypes',
    spacingsTokens: 'android/compose/spacingsTokens',
    spacingsTokensTypes: 'android/compose/spacingsTokensTypes',
    typography: 'android/compose/typography',
  },
}

export const TRANSFORM_GROUP = 'tokens-android'
