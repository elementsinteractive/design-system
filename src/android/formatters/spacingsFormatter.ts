import { Formatter } from 'style-dictionary'
import { lowerCase } from 'lodash'
import { useTemplate } from '../../utils'
import { AndroidThemeOptions } from '../types'

export const spacingsFormatter =
  (template: string, header: string, options: AndroidThemeOptions): Formatter =>
  ({ dictionary }) => {
    const tokens = dictionary.allTokens
    const spacings = tokens.map(({ name, value }) => ({
      name: lowerCase(name.replace('Spacings', '')),
      value,
    }))

    return useTemplate(template)({
      header,
      spacings,
      packageName: options.packageName,
    })
  }
