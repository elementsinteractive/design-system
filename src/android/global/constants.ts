import { join } from 'path'

import { AndroidGlobalOptions } from '../types'

export const ColorTypeFilename = 'ColorTokens.kt'

export const SpacingTypeFilename = 'SpacingTokens.kt'

export const GLOBAL_TEMPLATES = (options: AndroidGlobalOptions) => ({
  colorsTypeTemplate: createColorsTypeTemplate(options),
  spacingsTypeTemplate: createSpacingsTypeTemplate(options),
  headerTemplate: join(__dirname, './templates/header.ejs'),
})

const createSpacingsTypeTemplate = (options: AndroidGlobalOptions) => ({
  source: join(__dirname, './templates/spacingTokens.ejs'),
  destination: `${options.destination}/${SpacingTypeFilename}`,
  filter: 'spacingsType',
  formatter: 'android/spacingsType',
  packageName: options.packageName,
})

const createColorsTypeTemplate = (options: AndroidGlobalOptions) => ({
  source: join(__dirname, './templates/colorTokens.ejs'),
  destination: `${options.destination}/${ColorTypeFilename}`,
  filter: 'colorsType',
  formatter: 'android/compose/colorsType',
  packageName: options.packageName,
})
