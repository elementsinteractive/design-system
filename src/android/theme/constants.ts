import { join } from 'path'
import { AndroidThemeOptions } from '../types'

export const PaletteFilename = 'Palette.kt'
export const SpacingFilename = 'Spacing.kt'
export const TypeFilename = 'Type.kt'
export const ColorTokensFilename = 'ColorTokens.kt'

export const THEME_TEMPLATES = (options: AndroidThemeOptions) => ({
  typographyTemplate: createTypographyTemplate(options),
  spacingsTemplate: createSpacingsTemplate(options),
  paletteTemplate: createPaletteTemplate(options),
  colorTokensTemplate: createColorTokensTemplate(options),
  headerTemplate: join(__dirname, './templates/header.ejs'),
})

const createTypographyTemplate = ({ theme }: AndroidThemeOptions) => ({
  source: join(__dirname, './templates/typographies.ejs'),
  destination: `${theme.typography.destination}/${TypeFilename}`,
  filter: 'typography',
  formatter: 'android/compose/typography',
  packageName: theme.typography.packageName,
})

const createSpacingsTemplate = (options: AndroidThemeOptions) => ({
  source: join(__dirname, './templates/base/spacings.ejs'),
  destination: `${options.theme.spacings.destination}/${SpacingFilename}`,
  filter: 'spacings',
  formatter: 'android/spacings',
  packageName: options.theme.spacings.packageName,
  sharedPackageName: options.globalPackageName,
})

const createPaletteTemplate = ({ theme }: AndroidThemeOptions) => ({
  source: join(__dirname, './templates/shared/palette.ejs'),
  destination: `${theme.palette.destination}/${PaletteFilename}`,
  filter: 'color',
  formatter: 'android/compose/palette',
  packageName: theme.palette.packageName,
  themeName: theme.name,
})

const createColorTokensTemplate = ({ theme }: AndroidThemeOptions) => ({
  source: join(__dirname, './templates/shared/colorTokens.ejs'),
  destination: `${theme.colors.destination}/${ColorTokensFilename}`,
  filter: 'color',
  formatter: 'android/compose/colorTokens',
  packageName: theme.colors.packageName,
  themeName: theme.name,
})
