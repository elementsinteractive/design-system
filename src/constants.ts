export const LOG_PREFIX = '[Design]'