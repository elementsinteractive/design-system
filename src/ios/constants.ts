import { join } from 'path'

export const TEMPLATES = (destination: string) => ({
  typographyTemplate: {
    source: join(__dirname, './templates/typographies.ejs'),
    destination: `${destination}/Fonts.swift`,
    filter: 'isTypography',
    formatter: 'ios/typography',
  },
  spacingsTemplate: {
    source: join(__dirname, './templates/spacings.ejs'),
    destination: `${destination}/Spacings.swift`,
    filter: 'isSpacing',
    formatter: 'ios/spacings',
  },
  colorsTemplate: {
    source: join(__dirname, './templates/colors.ejs'),
    destination: `${destination}/Colors.swift`,
    filter: 'color',
    formatter: 'ios/colors',
  },
  headerTemplate: join(__dirname, './templates/header.ejs'),
})
