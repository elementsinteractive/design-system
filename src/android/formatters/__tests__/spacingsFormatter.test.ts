import createDictionary from 'style-dictionary/lib/utils/createDictionary'
import createFormatArgs from 'style-dictionary/lib/utils/createFormatArgs'

import { TEMPLATES } from '../../constants'
import { options } from './constants'
import { PROPERTIES } from '../../../common/formatters/__tests__/constants'
import { spacingsFormatter } from '../spacingsFormatter'

describe('spacingsFormatter', () => {
  it('can format spacings tokens', () => {
    const dictionary = createDictionary({ properties: PROPERTIES.spacings })
    const formatterArgs = createFormatArgs({ dictionary, platform: {} })

    const result = spacingsFormatter(
      TEMPLATES.FILES.spacingsTokens,
      'header',
      options,
    )(formatterArgs)

    expect(result).toMatchSnapshot()
  })
})
