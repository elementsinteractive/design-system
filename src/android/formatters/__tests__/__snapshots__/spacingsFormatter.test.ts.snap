// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spacingsFormatter can format spacings tokens 1`] = `
"header
package nl.elements.designsystem

object Spacings : SpacingTokens {
  override val xxxs = 4
  override val xxs = 8
  override val xs = 12
}"
`;
